src/components/LionessProvider.js

```diff
@@ -37,6 +38,15 @@ class LionessProvider extends Component {
     this.gt = getGettextInstance(props.messages, props.locale, options)
   }
 
+  componentDidUpdate(prevProps) {
+    if (prevProps.locale !== this.props.locale) {
+      emit()
+    }
+    if (prevProps.messages !== this.props.messages) {
```

Comment:
Is it intentional to check for referential equality?

Reply:
It was intentional, yes! But now that you ask, I guess this would break if you're doing some mutable cray-cray stuff like

```js
const mutableMessages = {}

class App extends Component {
  componentDidUpdate() {
    mutableMessages.someVal = Date.now()
    // Force update <App> somehow
  }

  render() {
    return (
      <LionessProvider messages={mutableMessages}>
        {/* ... */}
      </LionessProvider>
    )
  }
}
```

Reply:
What do you think?

Reply:
I personally think from an end user perspective a deep comparison is more predictable but comes at the cost of speed.

Reply:
Haha, this is tricky...

Deep comparison is kind of default React behaviour, since it updates components lavishly if you don't make them pure yourself. But deep comparison – or more specifically mutability – can also lead to side effects and expensive and unnecessary re-renders.

I think let's leave it and write a note on it in the README. It's very easy to make the prop a new object if you want to make use of the update mechanism.